src/nupic/types/Sdr.cpp

```diff
@@ -360,6 +360,62 @@ namespace sdr {
         SDR::setDenseInplace();
     }
 
+    void SparseDistributedRepresentation::concatenate(const SDR &inp1, const SDR &inp2, UInt axis)
+        { concatenate({&inp1, &inp2}, axis); }
+
+    void SparseDistributedRepresentation::concatenate(vector<const SDR*> inputs, UInt axis)
+    {
+        // Check inputs.
+        NTA_CHECK( inputs.size() >= 2u )
+            << "Not enough inputs to SDR::concatenate, need at least 2 SDRs got " << inputs.size() << "!";
+        UInt axis_dim_sum = 0u;
+        for( const auto &sdr : inputs ) {
+            NTA_CHECK( sdr != nullptr );
+            NTA_CHECK( sdr->dimensions.size() == dimensions.size() )
+                << "All inputs to SDR::concatenate must have the same number of dimensions as the output SDR!";
+            for( auto dim = 0u; dim < dimensions.size(); dim++ ) {
+                if( dim == axis ) {
+                    axis_dim_sum += sdr->dimensions[dim];
+                }
+                else {
+                    NTA_CHECK( sdr->dimensions[dim] == dimensions[dim] )
+                        << "All dimensions except the axis must be the same!";
+                }
+            }
+        }
+        NTA_CHECK( axis < dimensions.size() );
+        NTA_CHECK( axis_dim_sum == dimensions[axis] )
+            << "TODO: This one needs an explanation!";
```

Comment:
yep. 
`NTA_CHECK( axis_dim_sum == /*this concat.*/dimensions[axis] )` ?

Reply:
so it'd be `NTA_CHECK( concat_size == /*this concat.*/dimensions[axis] ) << "Resulting concatenated size at [axis] must match sum of its parts (at dimension [axis])"`